packages/backend/src/graphql/mutations/create-step.ts

```diff
@@ -15,24 +18,41 @@ const createStep: MutationResolvers['createStep'] = async (
   return await Step.transaction(async (trx) => {
     await trx.raw('SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;')
 
-    if (input.connection?.id) {
-      // if connectionId is specified, verify that the connection exists and belongs to the user
-      const connection = await context.currentUser
-        .$relatedQuery('connections')
-        .findOne({ id: input.connection.id })
-      if (!connection) {
-        throw new BadUserInputError('Connection not found')
-      }
-    }
-
     // Put SELECTs in transaction just in case there's concurrent modification.
     const flow = await context.currentUser
-      .$relatedQuery('flows', trx)
+      .withAccessible({ type: 'flow', trx, requiredRole: 'editor' })
       .findOne({
         id: input.flow.id,
       })
       .throwIfNotFound()
 
+    checkFlowUpdatable(input.flow.updatedAt, flow.updatedAt)
```

Comment:
nice check! can also be an instance method flow.assertNotUpdatedSince(clientUpdatedAt)